packages/pharos/src/components/icon/pharos-icon.ts

```diff
@@ -59,20 +84,22 @@ export class PharosIcon extends PharosElement {
 
   protected override render(): TemplateResult {
     const size = this._getIconSize();
-
+    const accessibilityLabel = this.a11yTitle || this.description;
+    // Check accessibilityLabel length for backwards compatibility until description is removed
+    const hideIcon = this.a11yHidden || accessibilityLabel === '';
     return html`
       <svg
         xmlns="http://www.w3.org/2000/svg"
         version="1.1"
         viewBox="0 0 ${size} ${size}"
         class="icon"
         role="img"
-        aria-hidden=${this.description === ''}
-        aria-label=${this.description || ''}
+        aria-hidden=${hideIcon || nothing}
```

Comment:
I think there's better shorthand for this unless I'm missing something:

```suggestion
        ?aria-hidden=${hideIcon}
```

Reply:
[Here are the relevant portion of the docs](https://lit.dev/docs/templates/expressions/#boolean-attribute-expressions) that I think mean the attribute will indeed be removed.

Reply:
Okay, I think I might see what is happening. When using `?aria-hidden=${hideIcon}` I get a bunch of failing unit tests from other components. That matched what I saw before when trying it. I had thought it was because it was not removing the `aria-hidden` attribute when `hideIcon` it was false. The tests are actually failing because when `hideIcon` is set to true, it is rendering the svg as `aria-hidden=""` which the axe plugin we use doesn't seem to recognize as being hidden.

With the `nothing` syntax `aria-hidden` is either not present or set explicitly to `aria-hidden="true"` which axe does seem to recognize. I'll see if there is a bug report for axe about that or if it is expected and we can go from there, it does seem to be rendering correctly in the browser with either syntax.

Reply:
A lil bit more research, I think that `aria-hidden` needs to have an explicit value set per the [spec](https://www.w3.org/TR/wai-aria-1.1/#aria-hidden) because it's a state and not a boolean attribute. So would need to be "true", "false", or undefined. 

Reference: https://stackoverflow.com/questions/55061083/does-aria-hidden-need-a-value

Reply:
@brentswisher that doesn't jibe with the docs or my understanding...something else is perhaps wrong? Or maybe the thing you think is evaluating as falsy is actually still truthy in some way?

@sirrah-tam I think we're almost saying the same thing, but I propose that `aria-hidden` should not be present here unless its value is `true`; to follow the spec exactly the way you're saying would imply to me that every single element in the DOM for which `aria-hidden` can be set would need to have it explicitly set to `aria-hidden="false"` which isn't my understanding.

Reply:
Yeah, I think the root of this is the fact we are using aria-hidden as a boolean attribute when it isn't.

By using `?` we are explicitly telling Lit "This is a boolean attribute" 

From the [HTML spec](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes):
>The presence of a boolean attribute on an element represents the true value, and the absence of the attribute represents the false value.

So Lit thinks based on the value of `hideIcon` the `aria-hidden` attribute needs to either:
- be present
- not exist 

Importantly, it never sets the value to anything either way, it just includes it or not. 

This matches what I am seeing in storybook (Note the `data-hidden-value` which I added to debug the value of `hideIcon`):
<img width="570" alt="Screenshot 2023-11-29 at 1 40 29 PM" src="https://github.com/ithaka/pharos/assets/6653970/d45ed26a-f87c-4545-889f-b743d426d354">

This also causes our accessibility tests to fail in other components: when aria-hidden should be `aria-hidden="true"` it is rendered as just `aria-hidden`

By using `aria-hidden=${hideIcon || nothing}` Lit doesn't treat it as a boolean, so it is correctly outputting `aria-hidden="true"`

I think the mental complication for this is stemming from the fact I specified `a11yHidden` as a boolean property on the component, which translates into `aria-hidden` on the svg, which is not a boolean property.

Reply:
Yep, I think we're saying the same thing, and IMO it makes the most sense to only render `aria-hidden` if it's needed to be set to true. My previous message was mostly trying to say that we shouldn't expect that only having `aria-hidden` without an assigned value (`aria-hidden=""`) to be parsed as a truthy attribute. The Accessibility API will ignore it, and it's why the axe plugin was throwing errors. 

💯 ☝️  @brentswisher

Reply:
This makes sense. @brentswisher @sirrah-tam do we want to make `a11yHidden` align with the `aria-hidden` spec for convergence/consistency's sake, and keep the rendering line here as-is?

Reply:
The overlap of meaning of JavaScript boolean `true|false` values and `undefined` with the string values `"true"|"false"|"undefined"` makes this challenging. I note that in Button's pressed state we handle everything explicitly as a string, so that they're always truthy (`"true"|"false"|"undefined"` are all truthy, as strings). I feel like that is closest to authoring raw HTML as well; if what you type is what ends up in the browser, most attributes then become either boolean (that is, present or not) or string. 

In Button we codified this even more directly for its pressed state; note that its type is `'true' | 'false' | 'undefined' | undefined` (different than what you have). I feel that the spec is really forcing `aria-hidden` to be a thing whose value is really a string, which gets coerced to a boolean or to a "idk, let the browser decide" behavior. I would vote we stick to that because I'd rather people maintain a mental model of the web than add a layer of indirection and muddy understanding when using these. My two cents.

Reply:
![Pharos Icon a11y excalidraw](https://github.com/ithaka/pharos/assets/6653970/23643368-f439-4968-a90e-40e5491012b2)


I have worked with Mat a bit to come up with this and believe the most recent code matches these expectations. I can also reuse the [ariaHiddenState](https://github.com/ithaka/pharos/pull/628/files#diff-ab0c9cb8869e1f27ee4d0a3c3532056dfee9e98d452e8a8c4cf947f425331159R4) from the button PR once that gets merged into develop for consistency.

Reply:
This is very helpful, thank you for documenting!